src/thermal.cpp

```diff
@@ -151,25 +151,25 @@ ThermalScattering::calculate_xs(double E, double sqrtkT, int* i_temp,
 {
   // Determine temperature for S(a,b) table
   double kT = sqrtkT*sqrtkT;
-  int i;
-  if (settings::temperature_method == TemperatureMethod::NEAREST) {
-    // If using nearest temperature, do linear search on temperature
-    for (i = 0; i < kTs_.size(); ++i) {
-      if (std::abs(kTs_[i] - kT) < K_BOLTZMANN*settings::temperature_tolerance) {
-        break;
-      }
-    }
-  } else {
+  int i = 0;
+
+  if (kTs_.size() > 1) {
     // Find temperatures that bound the actual temperature
     for (i = 0; i < kTs_.size() - 1; ++i) {
```

Comment:
Correcting myself in the above comment, there _is_ a problem for temperatures above the loaded cross section range. `i` gets incremented one more time upon leaving the loop below, referencing an invalid interval in the `kTs_` vector. Updating the condition here ensures we use an index for the correct temperature interval for temperatures either in or above the highest interval and lets us skip an unnecessary loop iteration.

```suggestion
    for (i = 0; i < kTs_.size() - 1; ++i) {
```

Reply:
Using a `while` loop would solve this and the problem above -- I'll fix this. Thanks for the careful review!